delphi-checks/src/test/java/au/com/integradev/delphi/checks/FormFmxCheckTest.java

```diff
@@ -0,0 +1,110 @@
+/*
+ * Sonar Delphi Plugin
+ * Copyright (C) 2024 Integrated Application Development
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02
+ */
+package au.com.integradev.delphi.checks;
+
+import au.com.integradev.delphi.builders.DelphiTestUnitBuilder;
+import au.com.integradev.delphi.checks.verifier.CheckVerifier;
+import org.junit.jupiter.api.Test;
+
+class FormFmxCheckTest {
+  @Test
+  void testNormalClassWithoutDfmShouldNotAddIssue() {
+    CheckVerifier.newVerifier()
+        .withCheck(new FormFmxCheck())
+        .withSearchPathUnit(createFmxForms())
+        .onFile(
+            new DelphiTestUnitBuilder()
+                .appendDecl("type")
+                .appendDecl("  TFoo = class(TForm)")
+                .appendDecl("  end;"))
+        .verifyNoIssues();
+  }
+
+  @Test
+  void testFormWithDfmShouldNotAddIssue() {
+    CheckVerifier.newVerifier()
+        .withCheck(new FormFmxCheck())
+        .withSearchPathUnit(createFmxForms())
+        .onFile(
+            new DelphiTestUnitBuilder()
+                .appendDecl("uses")
+                .appendDecl("  FMX.Forms;")
+                .appendDecl("type")
+                .appendDecl("  TFoo = class(TForm)")
+                .appendDecl("  end;")
+                .appendImpl("{$R Foo.fmx}"))
+        .verifyNoIssues();
+  }
+
+  @Test
+  void testFrameWithDfmShouldNotAddIssue() {
+    CheckVerifier.newVerifier()
+        .withCheck(new FormFmxCheck())
+        .withSearchPathUnit(createFmxForms())
+        .onFile(
+            new DelphiTestUnitBuilder()
+                .appendDecl("uses")
+                .appendDecl("  FMX.Forms;")
+                .appendDecl("type")
+                .appendDecl("  TFoo = class(TFrame)")
+                .appendDecl("  end;")
+                .appendImpl("{$R Foo.fmx}"))
+        .verifyNoIssues();
+  }
+
+  @Test
+  void testFormWithoutDfmShouldAddIssue() {
+    CheckVerifier.newVerifier()
```

Comment:
As mentioned in the DFM check test, I think it would be good to have some testing to confirm that:

- form in a file with other non-resource directives adds an issue
- form in a file with a non-FMX resource directive adds an issue
- etc. 

That's the most complex logic in the check and it's not really being tested here.